thoth/prescriptions_refresh/handlers/image_analysis.py

```diff
@@ -0,0 +1,218 @@
+      - latest
+      - performance
+      - stable
+      - testing
+      runtime_environments:
+        operating_systems:
+        - name: {os_name}
+          version: {os_version}
+        python_version: =={python_version}
+        base_images:
+          not:
+          - {image}
+    match:
+      state:
+        resolved_dependencies:
+        {resolved_dependencies}
+    run:
+      stack_info:
+      - type: INFO
+        message: >-
+          Found predictive stack image that can be used with these dependencies
+        link: {link}
+      advised_manifest_changes:
+      - file: .thoth.yaml
+        patch:
+        - name: base_image
+          value: {link}
+"""
+
+# TODO: Use Isis-API to identify categories or representative packages per topic?
+
+_REPRESENTATIVE_PACKAGES_ML = ["keras", "scikit-learn", "torch", "tensorflow"]
+
+_REPRESENTATIVE_PACKAGES_NLP = [
+    "gensim",
+    "nltk",
+    "spacy",
+    "transformers"
+]
+
+_REPRESENTATIVE_PACKAGES_CV = ["opencv-python", "pillow", "pytesseract", "torchvision"]
+
+_REPRESENTATIVE_PACKAGES = _REPRESENTATIVE_PACKAGES_ML + _REPRESENTATIVE_PACKAGES_NLP + _REPRESENTATIVE_PACKAGES_CV
+
+USER_API_HOST = os.environ["THOTH_USER_API_HOST"]
+
+
+def _get_latest_image_analyzed_info(image: str) -> Optional[Dict[str, Any]]:
+    """Get latest image analyzed information."""
+    url = f"http://{USER_API_HOST}/api/v1/container-images"
+    response = requests.get(url, params={"image_name": f"{QUAY_URL}/thoth-station/{image}"})
+
+    if response.status_code == 200:
+        results = response.json()
+        return results[0]
+    else:
+        return None
+
+
+def _get_requirement_files_from_image_analysis(
+    package_extract_document_id: str,
+) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+    """Get requirements files from image analysis result."""
+    url = f"http://{USER_API_HOST}/api/v1/analyze"
+    response = requests.get(url, params={"analysis_id": package_extract_document_id})
+
+    if response.status_code == 200:
+        document = response.json()
+
+        # Get result and requirements files
+        result = document["result"]
+        pipfile_dict = result["aicoe-ci"].get("requirements")
+        pipfile_lock_dict = result["aicoe-ci"].get("requirements_lock")
+
+        return pipfile_dict, pipfile_lock_dict
+
+    elif response.status_code == 404:
+        _LOGGER.warning(f"Document {package_extract_document_id} does not exist.")
+        return None, None
+
+    else:
+        return None, None
+
+
+def _create_resolved_dependencies_section(pipfile_d: Dict[str, Any], pipfile_lock_d: Dict[str, Any]) -> str:
+    """Create resolved dependencies section for adviser unit."""
+    resolved_dependencies = ""
+
+    pipfile = Pipfile.from_dict(pipfile_d)
+    pipfile_lock = PipfileLock.from_dict(pipfile_lock_d)
+
+    required_packages = []
+
+    for package in pipfile.packages.packages:
+        if package in _REPRESENTATIVE_PACKAGES:
```

Comment:
I'm wondering if `_REPRESENTATIVE_PACKAGES` could be directly requirements stated in Pipfile. Are there any exceptions for which we do not want to recommend the base but we want to keep them in Pipfile?

Reply:
I though about using the Pipfile initially, but then I remembered our conversation: https://github.com/thoth-station/prescriptions/pull/17213#discussion_r713365554 and thought about stating directly the representative packages. 
wdyt?

and about using Isis API for representative packages for specific contexts?

Reply:
> I thought about using the Pipfile initially, but then I remembered our conversation: [thoth-station/prescriptions#17213 (comment)](https://github.com/thoth-station/prescriptions/pull/17213#discussion_r713365554) and decided about stating directly the representative packages. wdyt?

OK, I leave this decision up to you.

> and about using Isis API for representative packages for specific contexts?

It might deserve new prescription handler. 👍🏻

Reply:
> > I thought about using the Pipfile initially, but then I remembered our conversation: [thoth-station/prescriptions#17213 (comment)](https://github.com/thoth-station/prescriptions/pull/17213#discussion_r713365554) and decided about stating directly the representative packages. wdyt?
> 
> OK, I leave this decision up to you.

I'm still not entirely sure to be honest, but based on that discussion having the base image recommended when there is only jupyter-tensorboard does not really make sense maybe, but if the user asks for torch + opencv-python then make sense for that. (wrap)

But for spreading the voice, it definitely make sense to iterate over all pipfile, I can add also Boot prescription as well here that consider `OR` package from pipfile as you suggested!
> 
> > and about using Isis API for representative packages for specific contexts?
> 
> It might deserve new prescription handler. 👍🏻

Reply:
> > > I thought about using the Pipfile initially, but then I remembered our conversation: [thoth-station/prescriptions#17213 (comment)](https://github.com/thoth-station/prescriptions/pull/17213#discussion_r713365554) and decided about stating directly the representative packages. wdyt?
> > 
> > 
> > OK, I leave this decision up to you.
> 
> I'm still not entirely sure to be honest, but based on that discussion having the base image recommended when there is only jupyter-tensorboard does not really make sense maybe, but if the user asks for torch + opencv-python then make sense for that. (wrap)
> 
> But for spreading the voice, it definitely make sense to iterate over all pipfile, I can add also Boot prescription as well here that consider `OR` package from pipfile as you suggested!
> 
> > > and about using Isis API for representative packages for specific contexts?
> > 
> > 
> > It might deserve new prescription handler. 👍🏻

My bad that boot is already implemented I fogort :D So we are good I think!

Reply:
> I'm still not entirely sure to be honest, but based on that discussion having the base image recommended when there is only jupyter-tensorboard does not really make sense maybe, but if the user asks for torch + opencv-python then make sense for that. (wrap)

+1. It might make sense to have packages as environment variables - so they become configuration and will not be hard-coded (which would require new releases when we change our minds about packages).

Reply:
I will add that logic then and add the packages in the CM in thoth-application :)

Reply:
Added https://github.com/thoth-station/prescriptions-refresh-job/pull/97/commits/c9f57d10b911593935ee5542c8c8ad93aad23e44 @fridex 🍻